FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -29,14 +29,59 @@ public final class CoreDataFeedStore: FeedStore {
 	}
 
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		performAction { context in
+			do {
+				let fetchRequest = NSFetchRequest<ManagedCache>(entityName: ManagedCache.className())
+				if let managedCache = try context.fetch(fetchRequest).first {
+					let managedFeed = managedCache.feed.compactMap { $0 as? ManagedFeedImage }.map {
+						LocalFeedImage(id: $0.id, description: $0.imageDescription, location: $0.location, url: $0.url)
+					}
+					completion(.found(feed: managedFeed, timestamp: managedCache.timestamp))
+				} else {
+					completion(.empty)
+				}
+			} catch {
+				completion(.failure(error))
+			}
+		}
 	}
 
 	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
-		fatalError("Must be implemented")
+		performAction { context in
+			do {
+				let managedCache = try ManagedCache.newUniqueInstance(in: context)
+				managedCache.timestamp = timestamp
+				managedCache.feed = NSOrderedSet(array: feed.map {
+					let managedFeedImage = ManagedFeedImage(context: context)
+					managedFeedImage.id = $0.id
+					managedFeedImage.imageDescription = $0.description
+					managedFeedImage.location = $0.location
+					managedFeedImage.url = $0.url
+					return managedFeedImage
```

Comment:
Suggestion:
`managedFeedImage` can be shortened to `image` since there're no other images inside this closure
```suggestion
					let image = ManagedFeedImage(context: context)
					image.id = $0.id
					image.imageDescription = $0.description
					image.location = $0.location
					image.url = $0.url
					return image
```

Reply:
Modified to `image`.